compiler/shared_ast/print.ml

```diff
@@ -61,32 +61,28 @@ let tlit (fmt : Format.formatter) (l : typ_lit) : unit =
 
 let location (type a) (fmt : Format.formatter) (l : a glocation) : unit =
   match l with
-  | DesugaredScopeVar (v, _st) ->
-    Format.fprintf fmt "%a" ScopeVar.format_t (Marked.unmark v)
-  | ScopelangScopeVar v ->
-    Format.fprintf fmt "%a" ScopeVar.format_t (Marked.unmark v)
+  | DesugaredScopeVar (v, _st) -> ScopeVar.format_t fmt (Marked.unmark v)
+  | ScopelangScopeVar v -> ScopeVar.format_t fmt (Marked.unmark v)
   | SubScopeVar (_, subindex, subvar) ->
     Format.fprintf fmt "%a.%a" SubScopeName.format_t (Marked.unmark subindex)
       ScopeVar.format_t (Marked.unmark subvar)
 
 let enum_constructor (fmt : Format.formatter) (c : EnumConstructor.t) : unit =
-  Format.fprintf fmt "%a"
-    (Utils.Cli.format_with_style [ANSITerminal.magenta])
+  Utils.Cli.format_with_style [ANSITerminal.magenta] fmt
     (Format.asprintf "%a" EnumConstructor.format_t c)
 
 let rec typ (ctx : decl_ctx option) (fmt : Format.formatter) (ty : typ) : unit =
   let typ = typ ctx in
   let typ_with_parens (fmt : Format.formatter) (t : typ) =
-    if typ_needs_parens t then Format.fprintf fmt "(%a)" typ t
-    else Format.fprintf fmt "%a" typ t
+    if typ_needs_parens t then Format.fprintf fmt "(%a)" typ t else typ fmt t
```

Comment:
Are you planning to improve the handling of parenthesis during the printing in the future? We could have a primitive precedence score in place in `*_needs_parens` no?

Reply:
Indeed that's part of my plans ;)